esyi/Resolver.ml

```diff
@@ -129,7 +131,22 @@ let setOCamlVersion ocamlVersion resolver =
   resolver.ocamlVersion <- Some ocamlVersion
 
 let setResolutions resolutions resolver =
-  resolver.resolutions <- resolutions
+  (* First we set the resolutions *)
+  resolver.resolutions <- resolutions;
+  (* Then we set the usage of every resolution to 0 *)
+  List.iter
+    ~f:(fun r -> Hashtbl.add resolver.resolutionUsage r false)
+    (Resolutions.entries resolutions)
+  
+let getUnusedResolutions resolver =
+  Hashtbl.fold
+    (fun (res:Resolution.t) used unused -> if not used then (res.name)::unused else unused)
+    resolver.resolutionUsage
+    []
+
+(* This function increments the resolution usage count of that resolution *)
+let didUse resolver resolution =
```

Comment:
nit: let's go with a more descriptive name if we are keeping this function at the top level `markResolutionAsUsed` or something like that.